presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -390,7 +390,17 @@ protected ExchangeNode build()
 
     public JoinNode join(JoinNode.Type joinType, PlanNode left, PlanNode right, JoinNode.EquiJoinClause... criteria)
     {
-        return new JoinNode(idAllocator.getNextId(),
+        return join(joinType, left, right, Optional.empty(), criteria);
+    }
+
+    public JoinNode join(JoinNode.Type joinType, PlanNode left, PlanNode right, Expression filter, JoinNode.EquiJoinClause... criteria)
+    {
+        return join(joinType, left, right, Optional.of(filter), criteria);
+    }
+
+    private JoinNode join(JoinNode.Type joinType, PlanNode left, PlanNode right, Optional<Expression> filter, JoinNode.EquiJoinClause... criteria)
```

Comment:
I don't believe so. I'd even say the order here is more natural b/c there's one filter and many criteria, and the criteria can be expressed as a filter and are themselves an optimization.